core/src/main/java/io/questdb/PropServerConfiguration.java

```diff
@@ -1156,16 +1160,8 @@ public PropServerConfiguration(
                 this.lineTcpWriterWorkerSleepThreshold = getLong(properties, env, PropertyKey.LINE_TCP_WRITER_WORKER_SLEEP_THRESHOLD, 10_000);
                 this.symbolCacheWaitUsBeforeReload = getLong(properties, env, PropertyKey.LINE_TCP_SYMBOL_CACHE_WAIT_US_BEFORE_RELOAD, 500_000);
 
-                int ilpTcpWorkerCount;
-                if (cpuAvailable < 9) {
-                    ilpTcpWorkerCount = 0;
-                } else if (cpuAvailable < 17) {
-                    ilpTcpWorkerCount = 2;
-                } else {
-                    ilpTcpWorkerCount = 6;
-                }
-                this.lineTcpIOWorkerCount = getInt(properties, env, PropertyKey.LINE_TCP_IO_WORKER_COUNT, ilpTcpWorkerCount);
-                cpuUsed += this.lineTcpIOWorkerCount;
+                // tested on 4/32/48 core servers
+                this.lineTcpIOWorkerCount = getInt(properties, env, PropertyKey.LINE_TCP_IO_WORKER_COUNT, cpuAvailable > 16 ? cpuAvailable / 2 : cpuAvailable);
```

Comment:
@bluestreak01 @nwoolmer 
I was under the impression that we would disable the I/O pool when there are fewer than 16 cores. Wasn't that the case?

Reply:
shoot, i did not push last commit, good spot

Reply:
fixed in https://github.com/questdb/questdb/pull/4221/commits/b5a7e03da64f20ff578820a6e8c34c130617d95a